controllers/pipelines/state_handler.go

```diff
@@ -98,41 +118,47 @@ func (st *StateHandler[R]) onUnknown(
 			logger.Error(err, fmt.Sprintf("%s, failing resource", failureMessage))
 
 			return []Command{
-				*From(resource.GetStatus()).WithSynchronizationState(apis.Failed).
+				*From(resource.GetStatus()).
 					WithVersion(newVersion).
-					WithMessage(failureMessage),
+					WithSyncStateCondition(apis.Failed, transitionTime, failureMessage),
 			}
 		}
 
 		return []Command{
 			*From(resource.GetStatus()).
-				WithSynchronizationState(apis.Updating).
+				WithSyncStateCondition(apis.Updating, transitionTime, "").
 				WithVersion(newVersion),
 			CreateWorkflow{Workflow: *workflow},
 		}
 	}
 
 	logger.Info("empty state, creating resource")
+
 	workflow, err := st.WorkflowFactory.ConstructCreationWorkflow(provider, providerSvc, resource)
 
 	if err != nil {
 		failureMessage := workflowconstants.ConstructionFailedError
 		logger.Error(err, fmt.Sprintf("%s, failing resource", failureMessage))
 
+		cmd := From(resource.GetStatus()).
+			WithSyncStateCondition(apis.Failed, transitionTime, failureMessage).
+			WithVersion(newVersion)
+
 		return []Command{
-			*From(resource.GetStatus()).WithSynchronizationState(apis.Failed).
-				WithVersion(newVersion).
-				WithMessage(failureMessage),
+			*cmd,
 		}
 	}
 
-	return []Command{
-		SetStatus{
-			Status: pipelineshub.Status{
-				Version:              newVersion,
-				SynchronizationState: apis.Creating,
-			},
+	status := SetStatus{
+		Status: pipelineshub.Status{
+			Version: newVersion,
 		},
+	}
+
+	status.WithSyncStateCondition(apis.Creating, transitionTime, "")
+
+	return []Command{
+		status,
```

Comment:
Minor naming suggestion: At the beginning of `stateTransition`, we create a `SetStatus` and call the variable `setStatus`. Here, we call it `cmd` first and then `status`. Is there a reason we didn't settle on one of these everywhere?